dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -1833,7 +1833,6 @@ Spdp::SpdpTransport::open()
 #endif
 
   local_sender_ = DCPS::make_rch<SpdpMulti>(reactor_task_.interceptor(), outer_->config_->resend_period(), ref(*this), &SpdpTransport::send_local);
-  local_sender_->enable(TimeDuration());
```

Comment:
Is this enabled by default?

Reply:
Hmm, I was seeing the participant message twice so I assumed yes, but the code says otherwise. Is something else writing the local announcement? If so, this can be enabled with the actual delay instead of empty TimeDuration.

Reply:
For platforms with a network config monitor, it writes the announcement after it joins the multicast group.

Reply:
What platforms won't have a network config monitor? Do I just need to deal with the fact that there will sometimes be a duplicate SPDP announcement?

Reply:
Linux.  Mac.  You should always be prepared to handle a duplicate SPDP announcement.